src/components/Widgets/Sabnzbd.vue

```diff
@@ -0,0 +1,212 @@
+    <div class="info-wrap">
+      <p class="info-line" v-for="sabznbd in sabnzbdQueue" :key="sabznbd.label">
+        <em :class="`fas fa-${sabznbd.status}`"></em>
+        <span class="lbl">{{ sabznbd.filename }}</span>
+        <span class="lbl">{{ sabznbd.percentage }}%</span>
+      </p>
+    </div>
+  </div>
+  <!-- Show/ hide toggle button for Queue-->
+  <p class="more-details-btn" @click="toggleQueue" v-if="sabnzbdQueue.length > 0">
+    {{ showQueue ? "Hide Queue" : "Show Queue" }}
+  </p>
+</div>
+</template>
+
+<script>
+import WidgetMixin from '@/mixins/WidgetMixin';
+
+export default {
+  mixins: [WidgetMixin],
+  data() {
+    return {
+      download_speed: null,
+      status: null,
+      sabnzbdDetails: [],
+      showQueue: false,
+      sabnzbdQueue: [],
+    };
+  },
+  mounted() {
+    this.checkProps();
+  },
+  computed: {
+    endpoint() {
+      const { apiKey, sabnzbdUrl } = this.options;
+      return `${sabnzbdUrl}/sabnzbd/api?output=json&apikey=${apiKey}&mode=queue`;
+    },
+  },
+  methods: {
+    /* Reads the kbpersec value of the server status, converts to mb/s if over 1024 kb. */
+    processKBperSec(kbpersec) {
+      if (kbpersec <= 1024) {
+        return `${Number(kbpersec).toFixed(0)} kb/s`;
+      } else {
+        return `${Number(kbpersec / 1024).toFixed(1)} mb/s`;
+      }
+    },
+    /* Reads the bool status output of the server status to append the correct icon */
+    processPaused(paused) {
+      if (paused === true) {
+        return 'pause';
+      } else {
+        return 'play';
+      }
+    },
+    /* Reads the string status output of the queue list to append the correct icon */
+    processPausedStr(paused) {
+      if (['Queued', 'Paused'].includes(paused)) {
+        return 'pause';
+      } else {
+        return 'play';
+      }
+    },
+    fetchData() {
+      this.makeRequest(this.endpoint).then(this.processData);
+    },
+    /* Fetches the Sabnzbd status, and processes results */
+    processData(data) {
+      this.download_speed = this.processKBperSec(data.queue.kbpersec);
+      this.status = this.processPaused(data.queue.paused);
+      if (!this.options.hideDetails) {
+        this.makeSabnzbdataData(data);
+      }
+      if (!this.options.hideQueue) {
+        this.makeSabnzbdataQueueData(data);
+      }
+    },
+    /* If showing Details, then Creates the object required */
+    makeSabnzbdataData(data) {
+      this.sabnzbdDetails = [
+        [
+          { label: 'Time Left', value: data.queue.timeleft },
+          { label: 'Queue', value: data.queue.noofslots },
+
+        ],
+        [
+          { label: 'Status', value: data.queue.status },
+          { label: 'Size Left', value: data.queue.sizeleft },
+        ],
+      ];
+    },
+    /* If showing Queue, Creates list of downloads that are in the sabnzbd list */
+    makeSabnzbdataQueueData(data) {
+      this.sabnzbdQueue = [];
+      let i = 0;
+      for (i; i < data.queue.slots.length; i += 1) {
```

Comment:
Again, this is totally fine as it is

But it can also be written in ES6 JS, using [`Array.forEach`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach), sometimes easier than a traditional for()

E.g.

```javascript
data.queue.slots.forEach((slot) => {
  this.sabnzbdQueue.push({
    status: this.processPausedStr(slot.status),
    filename: slot.filename.substring(0, 25),
    percentage: slot.percentage,
  });
});
```

Reply:
I haven't done a Loop like this before. But I will follow your guidance and apply the changes in the next revision :)